wire-ios/Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift

```diff
@@ -220,14 +219,13 @@ final class SettingsTableViewController: SettingsBaseTableViewController {
     }
 
     private func setupNavigationBar() {
-        navigationItem.rightBarButtonItem = navigationController?.closeItem()
+        navigationItem.rightBarButtonItem = UIBarButtonItem.closeButton(action: { [weak self] _ in
+            self?.dismiss(animated: true)
```

Comment:
_optional_: I usually prefer to explicitly call dismiss on the presenting view controller, even if the documentation mentions that this also works.

To be precise it should be:
```suggestion
            self?.presentingViewController?.dismiss(animated: true)
```

From the docs [UIViewController.dismiss(animated:completion:)](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621505-dismiss):

"Dismisses the view controller that was presented modally by the view controller." (so it's supposed to be called on the VC which is presenting another one)

"The presenting view controller is responsible for dismissing the view controller it presented." (VCs usually don't dismiss themselves)